hangman/game.py

```diff
@@ -1,13 +1,131 @@
 from .exceptions import *
-
+import random
 
 class GuessAttempt(object):
-    pass
-
+    
+    def __init__(self, character, hit=False, miss=False):
+        self.character = character
+        self.miss = miss
+        self.hit = hit
+        
+        if hit and miss:
+            raise InvalidGuessAttempt
+            
+        
+    def is_hit(self):
+        if self.hit:
+            return True
+        
+        if not self.hit: 
+            return False
+                
+        
+    def is_miss(self):
+        if self.miss: 
+            return True
+        
+        if not self.miss: 
+            return False
+            
 
 class GuessWord(object):
-    pass
+    def __init__(self, word_to_guess):
+         
+        self.answer = word_to_guess.lower()  
+        
+        if not word_to_guess:
+            raise InvalidWordException
+        else:
+            self.masked = len(word_to_guess)*"*"
+        
+
+    def perform_attempt(self,character):
+        character=character.lower()
+        if not character or len(character) > 1:  
+            raise InvalidGuessedLetterException
+
+
+        if character not in self.answer:
+            return GuessAttempt(character, hit= False, miss=True)
+                
+        
+        if character in self.answer:
+            '''Checks if user's guess matches any of the characters in answer word(string) by index. 
+            If there's a match, it is added to the string masked_word at the same index. Else it adds characters from self.masked.
+            Finally, it assigns the new masked_word to self.masked.'''
+            masked_word = ''
+            for index in range(0, len(self.answer)):
+                if  character == self.answer[index]:
+                    masked_word += character
+                else:
+                    masked_word += self.masked[index]
+
```

Comment:
try not to meddle with "index" if you can.. leave it to python `for` to handle it for you.. for example:

```python
for letter in self.answer:
   if character is letter:
      masked_word += character
   else:
      masked_word += letter
```